### DISCUSSION TITLE
Modifying number of anchors in Cascade RCNN causes AssertionError

### DISCUSSION THREAD

**Discussion by philwmanc** [open-mmlab/mmdetection#6391] [Q&A]
I've achieved some good detection performance using the Cascade RCNN model.  However, I'd like to try tuning the anchors to restrict the range of scales that are considered in order to reflect the fairly tight range of object sizes I'm interested in.  
Default anchor settings in config work fine:
        anchor_generator=dict(
            type='AnchorGenerator',
            scales=[8], 
            ratios=[0.5, 1.0, 2.0],
            strides=[4, 8, 16, 32, 64]),

However, if I try, for example:
        anchor_generator=dict(
            type='AnchorGenerator',
            scales=[8],
            ratios=[0.5, 1.0, 2.0],
            strides=[8, 16, 32]),  #this should be more representative of the range of object scales
This causes an AssertionError:

File "/mmdetection/mmdet/models/dense_heads/anchor_head.py", line 448, in loss
    assert len(featmap_sizes) == self.anchor_generator.num_levels

Does anyone know if it's possible to change the anchors like this, and if so what else needs to be done?  I haven't been able to find any documentation or previous discussions that address this.

Many thanks
Phil

**Comment by AndreaPi** [comment]
Maybe you could get more answers if you opened an issue of type "General Questions"?

https://github.com/open-mmlab/mmdetection/issues/new?assignees=&labels=&template=general_questions.md&title=

It sounds like an interesting question!

**Comment by Agito555** [comment]
You should change the FPN outputs too. 
In original setting, there will be 5 feature maps to be returned in the FPN. However, in the new setting, you just need 3 feature maps in the FPN.

**Reply by TonojiKiobya** [reply]
As Agito555 has put out, this is to say that the number of strides in the anchor_generator in the config should match the number of featuremaps your rpn_head gonna deal with. If it deals with 5 featuremaps, then there will be 5 strides, if it gonna deal with 1, there will be 1 stride. I need to test this myself though, but let us try!!!- One last educational point apart from the former Remember guys, in mmdetection, backbones and necks are dealing with tuples of featuremaps coming from different layers of backbones not normal feature maps
